PowerEditor/src/Notepad_plus.cpp

```diff
@@ -1446,64 +1439,65 @@ bool Notepad_plus::replaceInFiles()
 		_findReplaceDlg.setFindInFilesDirFilter(NULL, TEXT("*.*"));
 		_findReplaceDlg.getPatterns(patterns2Match);
 	}
-	vector<generic_string> fileNames;
-
-	getMatchedFileNames(dir2Search, patterns2Match, fileNames, isRecursive, isInHiddenDir);
 
 	Progress progress(_pPublicInterface->getHinst());
-	size_t filesCount = fileNames.size();
-	size_t filesPerPercent = 1;
+	progress.open(_findReplaceDlg.getHSelf(), TEXT("Replace In Files progress..."));
+	progress.setInfo(TEXT("Getting list of files, please wait..."));
 
-	if (filesCount > 1)
-	{
-		if (filesCount >= 200)
-			filesPerPercent = filesCount / 100;
-		progress.open(NULL, TEXT("Replace In Files progress..."));
-	}
+	vector<generic_string> fileNames;
+	getMatchedFileNames(dir2Search, patterns2Match, fileNames, isRecursive, isInHiddenDir, &progress);
 
-	for (size_t i = 0, updateOnCount = filesPerPercent; i < filesCount; ++i)
+	if (!progress.isCancelled())
 	{
-		if (progress.isCancelled()) break;
+		size_t filesCount = fileNames.size();

```

Comment:
`size_t filesCount` *should* be `const size_t filesCount`, because we're not adding/removing files.

So should `size_t filesPerPercent` (`const size_t filesPerPercent`)

Especially in a ~100+ line function.